src/main/java/uk/gov/hmcts/reform/civil/handler/tasks/TrialReadyCheckHandler.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.hmcts.reform.civil.handler.tasks;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.camunda.bpm.client.task.ExternalTask;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.civil.event.TrialReadyCheckEvent;
+import uk.gov.hmcts.reform.civil.service.search.TrialReadyCheckSearchService;
+
+import java.util.List;
+
+@Slf4j
+@RequiredArgsConstructor
+@Component
+public class TrialReadyCheckHandler implements BaseExternalTaskHandler {
+
+    private final TrialReadyCheckSearchService caseSearchService;
+    private final ApplicationEventPublisher applicationEventPublisher;
+
+    @Override
+    public void handleTask(ExternalTask externalTask) {
+        List<CaseDetails> cases = caseSearchService.getCases();
+        log.info("Job '{}' found {} case(s)", externalTask.getTopicName(), cases.size());
+
+        cases.forEach(caseDetails -> {
+            try {
+                applicationEventPublisher.publishEvent(new TrialReadyCheckEvent(caseDetails.getId()));
+            } catch (Exception e) {
+                log.error("Updating case with id: '{}' failed", caseDetails.getId(), e);
+            }
```

Comment:
A question here:
is it intentional that we log the exception and then swallow it continuing to the next one or are we meant to re-throw it?

Reply:
Good question - the other scheduler handlers were just logging, so I followed that pattern.
However, some other handlers are explicitly throwing errors, so perhaps it would be good to follow that example?
And then we should probably change the old handlers as well to bring everything into line.

Reply:
I'm inclined to throw the error, unless there's a proven reason to swallow it. Should that reason arise in the future, we can always change the code, whereas swallowing the error makes for unstable systems that don't get diagnosed.

Please make the change.

Reply:
Actually: leave that with me and keep focusing on your current ticket. I'll be putting up a commit to throw that exception momentarily.

Reply:
Okay, we'll leave it as is and I'll pick it up separately.